helix-term/src/commands.rs

```diff
@@ -3786,7 +3787,16 @@ fn paste_clipboard_impl(
     let (view, doc) = current!(editor);
     match editor.clipboard_provider.get_contents(clipboard_type) {
         Ok(contents) => {
-            paste_impl(&[contents], doc, view, action, count, editor.mode);
+            let has_equal_contents = editor.last_clipboard[clipboard_type as usize]
+                .join(doc.line_ending.as_str())
+                .eq(&contents);
```

Comment:
Instead of joining `last_clipboard` here (which allocats and therefore causes some overhead) you could iterate over it and compare `contents` on the fly. That would roughly look like the following:

```suggestion
	let mut pos = contents;
	let mut last_clipboard = editor.last_clipboard[clipboard_type as usize].iter();
	let mut has_equal_contents = true;
	if let Some(first) = last_clipboard.next() {
		if pos.starts_with(first) {
			pos = &pos.slice(first.len()];
			for selection in last_clipboard {
				if pos.starts_with(doc.line_ending.as_str()) && pos[1..].starts_with(selection) {
					pos = &pos[1 + selection.len()..];
				}else{
					has_equal_contents = false;
					break;
				}
			}
		}
	} else {
		has_equal_contents = false
	}
```

this is just something I came up quickly to illustrate what I mean so the implementation might be buggy/not compile/is not super idiomatic but I hope you get the idea

Reply:
I did some test with your idea and you are correct, it is much faster when there is large amount of data.
So, I just push a new commit to implement the idea. It uses a custom iterator for better readability, but the overall logic should remain the same.
Please take a look.